recyclerview/src/main/java/com/sagar/recyclerview/RecyclerView.kt

```diff
@@ -0,0 +1,55 @@
+package com.sagar.recyclerview
+
+import androidx.recyclerview.widget.RecyclerView
+import androidx.recyclerview.widget.RecyclerView.SCROLL_STATE_IDLE
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.channels.awaitClose
+import kotlinx.coroutines.flow.callbackFlow
+import kotlinx.coroutines.suspendCancellableCoroutine
+import kotlin.coroutines.resume
+
+/**
+ * method that takes care of first time notifying that the recyclerview scroll has ended
+ * use case may include the purpose where we do some user interaction after initial scroll
+ */
+@ExperimentalCoroutinesApi
+suspend fun RecyclerView.awaitScrollEnd() =
+    suspendCancellableCoroutine<Int> { continuation ->
+        // create anonymous object for recyclerview
+        val listener = object : RecyclerView.OnScrollListener() {
+            override fun onScrollStateChanged(recyclerView: RecyclerView, newState: Int) {
+                if (newState == SCROLL_STATE_IDLE) {
+                    // removing the listener so we don't leak the the continuation coroutine
+                    recyclerView.removeOnScrollListener(this)
+                    // Resume the coroutine once the scroll is idle
+                    continuation.resume(SCROLL_STATE_IDLE)
```

Comment:
Why can't we just pass `newState` here? Not sure `newState` will only have `SCROLL_STATE_IDLE` or other values as well.

Reply:
We can remove `SCROLL_STATE_IDLE` here as well.